### DISCUSSION TITLE
Use of this library with Isaac-sim

### DISCUSSION THREAD

**Discussion by zhangsanfeng86** [NVlabs/curobo#9] [Software Q&A] [answered]
Hello,

The library is great and  awesome!
The standalone example works well in curobo docker container.
How can I  run the issac-sim example  with graph UI interface?

Thank you in advance!

Peter

**Comment by balakumar-s** [comment]
We are currently working on a dockerfile for isaac sim examples. In the meantime, you can try the isaac sim examples by installing curobo natively following instructions here: https://curobo.org/source/getting_started/1_install_instructions.html#install-for-use-in-isaac-sim

**Reply by zhangsanfeng86** [reply]
thank you!

Look forward the dockerfile!

**Reply by zhangsanfeng86** [reply]
Following instructions：https://curobo.org/source/getting_started/1_install_instructions.html#install-for-use-in-isaac-sim
I install curobo in docker container "nvcr.io/nvidia/isaac-sim:2022.2.1"
But got Error as follow:

![image](https://github.com/NVlabs/curobo/assets/73547638/d17f0f29-dc94-4ca2-850b-5bf0f15e7992)

**Reply by balakumar-s** [reply]
Those instructions will not work for an isaac sim docker. They will only work for a native install.

**Reply by zhangsanfeng86** [reply]
OK,
Thank you for your reply!

After a native install , when run "examples/isaac_sim/ik_reachability.py" ,  got Error:
<img width="1440" alt="image" src="https://github.com/NVlabs/curobo/assets/73547638/16179343-13e6-4b21-85e8-537fca61aea3">

**Reply by balakumar-s** [reply]
Did a window popup with isaac sim?

**Reply by zhangsanfeng86** [reply]
yes,   
but no robot. ,only one cube

**Reply by balakumar-s** [reply]
It's because of not having git lfs. Can you try these instructions: https://github.com/NVlabs/curobo/issues/10#issuecomment-1788363448

**Reply by zhangsanfeng86** [reply]
and I found another question:
When start a new terminal, then conda activate curobo,   run "python examples/isaac_sim/motion_gen_reacher.py ",  got this error <img width="854" alt="image" src="https://github.com/NVlabs/curobo/assets/73547638/eac5edeb-fd9f-467d-b60f-786a44e0b412">

**Reply by balakumar-s** [reply]
Is this conda environment separate from isaac sim? 
Let's try with the native python from isaac sim first to make sure everything works.

**Reply by zhangsanfeng86** [reply]
Great!
It work!
run "examples/isaac_sim/motion_gen_reacher.py "
<img width="591" alt="image" src="https://github.com/NVlabs/curobo/assets/73547638/8fe66172-47dd-4c1f-b65c-741c52e7ea5c">

**Reply by zhangsanfeng86** [reply]
But run "examples/isaac_sim/simple_stacking.py "
got error:
<img width="1296" alt="image" src="https://github.com/NVlabs/curobo/assets/73547638/b4804fba-d88d-4912-941b-682609cd6b49">

**Reply by balakumar-s** [reply]
It looks like it's searching in the conda env for some libraries, is conda disabled?

**Reply by zhangsanfeng86** [reply]
conda activate 
<img width="318" alt="image" src="https://github.com/NVlabs/curobo/assets/73547638/d0385171-2339-450b-93c4-bcbf90f714e6">

**Reply by balakumar-s** [reply]
Sorry, there might be some confusion here.

Can you try with `conda deactivate` and using omni_python ?

**Reply by zhangsanfeng86** [reply]
ok i'll try

**Reply by zhangsanfeng86** [reply]
it doesn't work
<img width="1175" alt="image" src="https://github.com/NVlabs/curobo/assets/73547638/0ef68b00-6848-4b11-8f13-ae14c261139c">

**Reply by balakumar-s** [reply]
This error only happens with `omni_python examples/isaac_sim/simple_stacking.py` and not with any other examples?

For example:
 `omni_python examples/isaac_sim/motion_gen_reacher.py` works?

**Reply by zhangsanfeng86** [reply]
it work : "omni_python examples/isaac_sim/motion_gen_reacher.py"
![image](https://github.com/NVlabs/curobo/assets/73547638/5bed27bc-46ec-4fb8-838f-df3d13101f81)

**Reply by zhangsanfeng86** [reply]
"multi_arm_reacher.py" also work
<img width="810" alt="image" src="https://github.com/NVlabs/curobo/assets/73547638/3761e826-578d-468d-87c2-45630585fe4e">

**Reply by balakumar-s** [reply]
I am not sure why the error happens only with that example. 

I will push an isaac sim docker which will have everything working before the end of this week.

**Reply by balakumar-s** [reply, marked as answer]
I think I found the issue. Can you comment out this line: https://github.com/NVlabs/curobo/blob/f2eb5f937afef065214daab7e09a56e8f7880176/examples/isaac_sim/simple_stacking.py#L67

And try again?

**Reply by balakumar-s** [reply]
This fixed the issue on my end. I will push a fix for this as well. Thanks for debugging with me.
